src/tmpfiles/tmpfiles.c

```diff
@@ -1126,6 +1128,85 @@ static int parse_acls_from_arg(Item *item) {
 }
 
 #if HAVE_ACL
+static int set_acl_cond_exec(const char *path, acl_t cond_exec, acl_t *ret, bool isdir) {
```

Comment:
as per coding style, `ret` and `ret_xyz` is how return parameters should be called. But this here is also an input parameter. Hence it shouldn#t be called that way.

That said: it really should not be an input parameter, it should really be output only. with the curretn code you actually *modify* the non-exec ACL object that is stored in the `Item` object, but such an object is possibly applied to a myriad of inodes. Thus, you are modifying the ACL the first time you find a matching inode, and then it will be applied for all future inodes too. and that's just wron.g

Reply:
instead of passing in an opaque bool i'd just pass in `st_mode` as is, and place the IS_DIR() inside of this function. i.e. encapsulate the special handling for dirs locally instead of indicating that in the function prototype already